watch_test.go

```diff
@@ -50,6 +50,30 @@ func TestWatch(t *testing.T) {
 	}
 }
 
+func TestWatchRedirectPreservesHeader(t *testing.T) {
+	t.Parallel()
+
+	redirectServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		assert.Equal(t, []string{"text/event-stream"}, req.Header["Accept"])
```

Comment:
Making sure that when we do a watch, the text/event-stream header appears in the request to the redirected server.

I think this covers the change, since when I just returned nil from the added code, the test failed.